rfd/0087-access-request-notification-routing.md

```diff
@@ -0,0 +1,207 @@
+  while most plugins use a role_to_recipient map in the plugin config. Users leveraging both PagerDuty and Slack plugins 
+  have to deal with two configuration languages;
+- Current Pagerduty request routing applies the same rule to all requests from the same role, regardless of the
+  requested role. All access requests don't have the same severity, users wanting to route access requests differently
+  have to create multiple roles. Depending on the company size and structure cardinality can become a problem.
+  See https://github.com/gravitational/teleport-plugins/issues/597 ; and
+- Access request routing map baked in the plugin deployment:
+  - Will become huge for big orgs.
+  - Requires redeploying each time a change is done.
+  - Does not allow to route requests regarding of the requestor.
+
+## Details
+
+### User story
+
+Bob is a developer. He deploys code using CI/CD pipelines. He can request access to prod in read-only mode and dev in read-write mode for debugging. In case of incident he can request prod read-write access.
+Alice is the lead-dev, she grants regular access requests through MsTeams during open hours and can also approve urgent read-write requests.
+Alice should not be paged each time Bob needs to debug something during open hours, but in case of incident Bob needs immediate access and Alice should be paged.
+
+In Teleport terms:
+
+- the role `developer` can request roles `dev-rw`, `prod-ro` and `prod-rw`;
+- the role `lead-developer` can accept requests;
+- only `prod-rw` access requests should trigger a PagerDuty incident; and
+- `dev-rw` and `prod-ro` access requests should trigger a MsTeams message.
+
+### Terminology
+
+Definitions:
+- A target is the pair: (`plugin`, `recipient`);
+- `plugin` is a string describing which plugin should pick up this target;
+  By default each plugin are listening for their own name (`slack` for the slack plugin,
+  `msteams` for the Microsoft Teams one, ...). But the plugin name can be configurable
+  in the plugin configuration. This allows support for multiple instances of the same
+  plugin (e.g. many orgs have multiple slack workspaces, they can run 1 plugin per workspace); and
+- `recipient` is a string describing the access request recipient. Different kinds of recipients
+  are supported depending on the plugin (user name, channel name, service name, user id,
+  channel id, URL, ...).
+
+### New Resource: `access_routing_rule`
+
+This proposal introduces a new resource to add routing information to newly created access-requests.
+
+The motivation behind adding a new resource, opposed to add new fields in `role` is to
+limit the `role` spec inflation and allow more flexibility to create role-agnostic routing rules.
+
+Routing rules are evaluated on access-request creation and will add elements to a new field in
+access-requests: `targets`.
+
+An example access_routing_rule would look like:
+
+```yaml
+kind: access_routing_rule
```

Comment:
Nit: Not sure if it's too verbose but `access_request_routing_rule` would probably be a bit more accurate name for this resource.

Reply:
done